src/r_tipping/RateSystem.jl

```diff
@@ -0,0 +1,98 @@
+# we consider the ODE dxₜ/dt = f(xₜ,λ(rt))
+# here λ = λ(t) ∈ Rᵐ is a function containing all the system parameters 
+
+# We ask the user to define: 
+#  1) a ContinuousTimeDynamicalSystem that should be investigated and
+#  2) a protocol for the time-dependent forcing with the struct RateProtocol
+
+# Then we give back the ContinuousTimeDynamicalSystem with the parameter 
+# changing according to the rate protocol
+"""
+    RateProtocol
```

Comment:
I would recommend to rename this to `RateConfig`. Config, from configuration, is the name give typically in Julia to such composite types whose purpose is to configure some functionality. I like a lot that this is a dedicated struct, instead of keyword arguments to a function. But I believe naming it Config also makes it more harmonious with DynamicalSYstems.jl which follows this convention in many packages.